src/hipscat_import/catalog/margin_cache_arguments.py

```diff
@@ -0,0 +1,94 @@
+            the margin data before doing more precise boundary checking. this value
+            must be greater than the highest order of healpix partitioning in
+            the source catalog. if `margin_order` is left default or set to -1, then
+            the `margin_order` will be set dynamically to the highest partition
+            order plus 1.
+        overwrite (bool): if there is existing data at the `catalog_path`, should
+            we overwrite and create a new catalog.
+        pixel_threshold (int): maximum number of rows for a single resulting pixel.
+            we may combine hierarchically until we near the `pixel_threshold`
+        mapping_healpix_order (int): healpix order to use when mapping. will be
+            `highest_healpix_order` unless a positive value is provided for
+            `constant_healpix_order`.
+        debug_stats_only (bool): do not perform a map reduce and don't create a new
+            catalog. generate the partition info.
+        tmp_path (str): path for storing intermediate files
+        `_tmp_dir` (str): base directory provided by the caller for temporary files.
+        progress_bar (bool): if true, a tqdm progress bar will be displayed for user
+            feedback of map reduce progress.
+        dask_tmp (str): directory for dask worker space. this should be local to
+            the execution of the pipeline, for speed of reads and writes.
+        dask_n_workers (int): number of workers for the dask client
+        dask_threads_per_worker (int): number of threads per dask worker.
+    """
+
+    margin_threshold: float = 5.0
+    margin_order: int = -1
+
+    input_catalog_path: str = ""
+    margin_output_path: str = ""
+
+    def __post_init__(self):
+        super().__post_init__()
+        self._check_arguments()
+
+    def _check_arguments(self):
+        if not file_io.does_file_or_directory_exist(self.input_catalog_path):
+            raise FileNotFoundError("input_catalog_path not found on local storage")
+
+        meta_pointer = paths.get_catalog_info_pointer(self.input_catalog_path)
+        catalog_info = CatalogInfo.read_from_metadata_file(meta_pointer)
+
+        info_pointer = paths.get_partition_info_pointer(self.input_catalog_path)
+        pixels = PartitionInfo.read_from_file(info_pointer)
+
+        self.catalog = Catalog(
+            catalog_info=catalog_info,
+            pixels=pixels,
+            catalog_path=self.input_catalog_path
+        )
+
+        partition_stats = self.catalog.get_pixels()
+        highest_order = np.max(partition_stats["Norder"].values)
+        margin_pixel_k = highest_order + 1
+        if self.margin_order > -1:
+            if self.margin_order < margin_pixel_k:
+                # pylint: disable=line-too-long
+                raise ValueError(
+                    "margin_order must be of a higher order than the highest order catalog partition pixel."
+                )
+                # pylint: enable=line-too-long
```

Comment:
```suggestion
                raise ValueError(
                    "margin_order must be of a higher order than "
                    "the highest order catalog partition pixel."
                )
```

And elsewhere. I don't know if there's an official recommendation, but I find the line breaks a lot easier to read, and results in fewer overall lines of code.